### ISSUE TITLE
Ability to assign roles on a subscription level

### ISSUE THREAD

**Issue by rikunarhi-cloud2** [Azure/terraform-azurerm-caf-enterprise-scale#420]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently its not possible to add role assignments on a subscription level.

The role assignment list seems to just be an empty list currently. Are there any plans to use this in the future?

```hcl

es_role_assignments_by_subscription = local.empty_list

```

The problem comes from us having multiple subscriptions in the same landing zone management group. The subscriptions have different permissions depending on what the person is working on.

#### Describe the solution you'd like

I would like to have the ability to assign roles on a subscription level in addition to management group level.

**Comment by krowlandson** [comment]
Thank you for your question @rikunarhi-cloud2.

This was added into the code when I first created the module as we were considering this as a future enhancement. Unfortunately limitations in the provider restrict us to only being able to deploy resources to Subscriptions configured directly in a provider block (hence the need to specify 3 provider blocks into the module for `core`, `management` and `connectivity`).

As a customer will likely be working with an unknown number of subscriptions, there isn't currently a realistic way for us to offer this functionality using the `azurerm` provider. As such we don't have plans to implement this feature.

Instead, we recommend customers create subscription scoped policies and roles using resource declarations outside of the module.
